specs/architecture.md

```diff
@@ -0,0 +1,30 @@
+Architecture
+===
+
+- [Architecture](#architecture)
+- [Common Terms and Expressions](#common-terms-and-expressions)
+- [System Architecture](#system-architecture)
+
+# Common Terms and Expressions
+
+| name              | description                                                                                                                                                                      |
+| ----------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| App (application) | Alternate name for "[virtual sidechain](https://arxiv.org/abs/1905.09274)." LazyLedger apps are sidechains that post all their data onto the LazyLedger chain to share security. |
+| Transaction       | Requests that modify the consensus-critical state (validator balances and statuses).                                                                                             |
+| Message           | Requests that are executed by non-consensus-critical apps.                                                                                                                       |
+
+# System Architecture
+
+![fig: Block data structures.](figures/block_data_structures.svg)
```

Comment:
We still need a single state root in the top-level header, which should replace the validators roots. That's orthogonal to the block data roots, but only a single root is required so that intermediate state roots in the block data can be linked to it for fraud proofs.